src/main/java/Duke.java

```diff
@@ -1,10 +1,152 @@
+        parser = new Parser();
+    }
+
+    /**
+     * Main
+     */
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+        new Duke().run();
+    }
+
+    /**
+     * Take in command
+     */
+    public void run() {
+        ui.sayHi();
+        Scanner myScanner = new Scanner(System.in);
+        while(true) {
+            String cmd = myScanner.nextLine();
+            if(cmd.equals("bye")) {
+                ui.sayBye();
+                break;
+            }
+            else if(cmd.equals("list")) {
+                System.out.println("Here are the tasks in your list:");
+                for(int i = 1; i <= tasks.getSize(); ++i) {
+                    System.out.println(i + "." + tasks.get(i - 1).getStatus());
+                }
+            }
+            else if(cmd.length() >= 4 && cmd.substring(0, 4).equals("done")) {
+                int c = 0;
+                for(int i = 5; i < cmd.length(); ++i) {
+                    c = c * 10 + cmd.charAt(i) - '0';
+                }
+                System.out.println("Nice! I've marked this task as done:");
+                tasks.get(c - 1).done();
+                System.out.println(tasks.get(c - 1).getStatus());
+            }
+            else if(cmd.length() >= 4 && cmd.substring(0, 4).equals("todo")) {
+                try {
+                    checkCmd(cmd, 4, "☹ OOPS!!! The description of a todo cannot be empty.");
+                    String getName = cmd.substring(5);
+                    Todo tmp = new Todo(getName);
+                    System.out.println("Got it. I've added this task: ");
+                    System.out.println("  " + tmp.getStatus());
+                    tasks.add(tmp);
+                    System.out.println("Now you have " + tasks.getSize() + " tasks in the list.");
+                }
+                catch (DukeException ex) {
+                    System.out.println(ex.getMessage());
+                }
+            }
+            else if(cmd.length() >= 8 &&cmd.substring(0, 8).equals("deadline")) {
+                try {
+                    checkCmd(cmd, 8, "☹ OOPS!!! The description of a deadline cannot be empty.");
+                    String getName = parser.getNameBy(cmd);
+                    String getDeadline = parser.getDeadlineBy(cmd);
+                    getDeadline = formatDate(getDeadline);
+                    Deadline tmp = new Deadline(getName, getDeadline);
+                    System.out.println("Got it. I've added this task: ");
+                    System.out.println("  " + tmp.getStatus());
+                    tasks.add(tmp);
+                    System.out.println("Now you have " + tasks.getSize() + " tasks in the list.");
```

Comment:
I appreciate how clean your code is in the try blocks! But maybe you could include more spacings and comments between different segments of the code so that it is easier to capture the main gist of your code?